### DISCUSSION TITLE
⏳ 🔗 Make global widgets content editable without unlinking

### DISCUSSION THREAD

**Discussion by nmbrthirteen** [elementor/elementor#22203] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

I've saved the global widget and used this widget on too many pages. on a couple of pages, I have to use this widget with different content but the same style. if I want to keep the style of the widget but change only the content, it isn't possible with elementor. imagine including this widget on 100+ pages and editing them by hand because if I need to change content I have to unlink it or duplicate it as the new widget which is additional work.

### Describe the solution you'd like

Figma components are the perfect example of this. you're saving styles of different design parts and using them in this way. it is possible to edit content which is very great, productive and time-saving.

### Describe alternatives you've considered

I can't find such a thing anywhere. I don't know if I missed something, all I know is that Elementor doesn't have this feature.

### Additional context

This problem is much bigger for big websites with hundreds of pages and templates. small ones don't have this issue.

**Comment by stannvanheusden** [comment]
I was about to open a discussion/issue about the same thing. I think [Figma components](https://help.figma.com/hc/en-us/articles/360038662654-Guide-to-components-in-Figma#:~:text=Components%20are%20elements%20you%20can,icons%2C%20layouts%2C%20and%20more.) is a great example indeed.

It would be a great functionality improving the usability for end-users. At this moment it's only possible to set one global button style for example, so for secondary buttons you need to style the element, which requires copy/paste style for clients. That's a hassle. 

If you have a component library (maybe better than "Global Widgets"?) featuring default settings, it would super charge editing sites. Maybe it could display icons of the rendered component instead of the SVG icons currently in the editor. For example:

![elementor-component](https://user-images.githubusercontent.com/5797196/236791871-743e6938-e07a-4625-99fa-c446ffc05025.png)



Thank you for opening this discussion. I hope the Elementor developers can work on this :-)!

**Comment by jpalusky** [comment]
Webflow has a feature like this called "Component Properties" that might be an inspiration for Elementor: https://university.webflow.com/lesson/components#how-to-define-component-properties

If this feature was added to Elementor Global Widgets I would be thrilled. Currently it is very difficult to maintain Elementor websites because you have to unlink your widgets every time you want to change anything. I'm considering migrating away from Elementor for this sole reason.

**Comment by nomadscode** [comment]
+1

I've changed recently to Elementor for some sites, and coming from the Divi theme builder, I'm surprised to find out that you can't have a global element in which you can only change certain parts like the text. 

For me, this kind of defeats the purpose of a global element. A common use case is a button; right now with Elementor I have to unlink all the button individually when I want to change the text, which means they won't get updated anymore in terms of styling. In other words, going through the complete website and update them individually, each time style changes. 

I'm fixing it with CSS now, but would be great to see this in the plugin itself.

**Comment by dogooddesignco** [comment]
I'm trialing Elementor considering moving to it from custom-coded sites and this is a total deal breaker. The inability to edit properties of global widgets means a site becomes unmaintainable. Of course I could write custom CSS for all my buttons but that defeats the purpose of using a page builder.

**Comment by theavic** [comment]
Can this be triaged soon?

**Comment by nmbrthirteen** [comment]
This is what I was talking about - https://www.youtube.com/watch?v=wTNfnf5mq4o 

WordPress will release this soon. I'm fully working with Elementor, and this feature is critically important to me. 

It's time to prioritize this problem, @nicholaszein.

**Reply by nomadscode** [reply]
This excactly. Thanks for sharing. 

I reckon it would be best to have this feature directly implemented into Elementor, but perhaps there is a way to utilize the new Partially Synced Patterns from WordPress inside Elementor.

Can we get a status update on this?

**Reply by nmbrthirteen** [reply]
They said this feature is outside the roadmap. I hope their plans will change with this 🤷‍♂️ Waiting for it

**Comment by L-U-C-K-Y** [comment]
This would immensely increase the appeal of Elementor. Framer and Figma work this way and speed up my workflow significantly only because of this feature. 

It would be great to create my blocks/templates and then reuse and override them on my site.

**Comment by Tchelet-Levi** [comment]
Definitely wished there was a feature like this myself. My workaround is to use classes and style them myself and only use Elementor for the content, but at that point what's the point you know?

**Reply by nmbrthirteen** [reply]
This feature will make working with Elementor a lot easier. When you have the same sections but different content on many pages, you can change the core component design, and all will change without modifying the content.

You will avoid custom css and classes, and this makes sense.

It is for big websites mainly, and for example, in React, you’re using a component-based structure, and you can pass whatever content you need without copying and pasting whole sections for no reason. It’s common practice, and we still don’t have it in Elementor.

I hope this will be implemented soon. Thanks for your comment

**Comment by min0k** [comment]
Please add this feature. Makes the difference between using Elementor and not using it.

**Comment by fanny-peneau** [comment]
I totally agree. Whether in design or development we use **components**. 
This makes it possible to maintain a consistent design system, which is currently impossible in Elementor. 

Yes you can code your own widget but it is very tedious.

As some suggest, the Webflow approach is what I have seen best 🚀

**Comment by ghost** [comment]
+1 Global Widgets need some overall love.

**Comment by actik** [comment]
+1 This feature is essential to create modern design systems.

**Comment by juuso-turunen** [comment]
+1 actually this would be very helpful and it feels mandatory

**Comment by chris-jut-so** [comment]
+1. Right now we're often custom building widgets for that. This would save so much time!

**Comment by neuronz** [comment]
ChatGPT4o gives its 2 cents worth on the issue:

"It would indeed be a great feature if Elementor allowed individual fields to be "unlinked" or excluded from the global settings, offering more granular control over global widgets. This would provide the flexibility to keep certain aspects like style and design consistent while allowing specific content, like text, to be customized per instance without fully unlinking the widget.

Elementor is always evolving, and user feedback plays a significant role in shaping new features. If you think this would be a valuable addition, you could consider submitting a feature request to the Elementor team. They might consider adding such functionality in future updates."

**Reply by nmbrthirteen** [reply]
This is the only way to submit feature requests. I did it over a year ago, but this discussion still awaits triage. I see how much they care about customer feedback and feature requests. I hope they will work on this feature soon

**Comment by agrbch** [comment]
Definitely a must-have feature!
